src/libYARP_OS/include/yarp/os/LockGuard.h

```diff
@@ -27,29 +33,43 @@ namespace yarp {
  * the LockGuard is destructed and the mutex is released.
  * The lock_guard class is non-copyable. 
  */
-class YARP_OS_API yarp::os::LockGuard {
+template <typename Lockable>
+class yarp::os::AbstractLockGuard {
 public:
     /**
      * Acquires ownership of the given mutex _mutex.
      * The behavior is undefined if _mutex is destroyed before the LockGuard object is.
      * @param _mutex the mutex which will be locked
      */
-    explicit LockGuard(yarp::os::Mutex& _mutex);
+    explicit AbstractLockGuard(Lockable& _lock);
     
     /**
      * destructs the LockGuard object, unlocks the underlying mutex
      */
-    ~LockGuard();
+    ~AbstractLockGuard();
     
 private:
     
     /** Copy constructor is disabled */
-    LockGuard(const LockGuard&);
+    AbstractLockGuard(const AbstractLockGuard&);
     
     /** Assignment operator is disabled */
-    LockGuard& operator=(const LockGuard&);
+    AbstractLockGuard& operator=(const AbstractLockGuard&);
     
-    yarp::os::Mutex& mutex; /*!< underlining mutex */
+    Lockable& lock; /*!< underlining mutex */
 };
 
+#ifndef YARP_LG_EXTERN
```

Comment:
I looked for this definition in that file, but I did not find it. I found the contrary (extern for windows, nothing for linux)